src/main/java/org/json/XML.java

```diff
@@ -445,6 +448,48 @@ private static boolean parse(XMLTokener x, JSONObject context, String name, XMLP
             }
         }
     }
+    /**
+     * This method removes any JSON entry which has the key set by XMLParserConfiguration.cDataTagName
+     * and contains whitespace as this is caused by whitespace between tags. See test XMLTest.testNestedWithWhitespaceTrimmingDisabled.
+     * @param jsonObject JSONObject which may require deletion
+     * @param config The XMLParserConfiguration which includes the cDataTagName
+     */
+    private static void removeEmpty(final JSONObject jsonObject, final XMLParserConfiguration config) {
+        if (jsonObject.has(config.getcDataTagName()))  {
+            final Object s = jsonObject.get(config.getcDataTagName());
+            if (s instanceof String) {
+                if (isStringAllWhiteSpace(s.toString())) {
+                    jsonObject.remove(config.getcDataTagName());
+                }
+            }
+            else if (s instanceof JSONArray) {
+                final JSONArray sArray = (JSONArray) s;
+                for (int k = sArray.length()-1; k >= 0; k--){
+                    final Object eachString = sArray.get(k);
+                    if (eachString instanceof String) {
+                        String s1 = (String) eachString;
+                        if (isStringAllWhiteSpace(s1)) {
+                            sArray.remove(k);
+                        }
+                    }
+                }
+                if (sArray.isEmpty()) {
+                    jsonObject.remove(config.getcDataTagName());
```

Comment:
Is there a reason for removing the key?

Reply:
The value at this key will always be empty as it was inadvertently added due to the whitespace in between tags as described above. We remove it as we don't want a key with an empty value

Reply:
Could an empty array have been valid XML text?

Reply:
I'm not sure if it is valid or not, but the reason it would be empty at this point is because we are removing all the entries as they are confirmed to have been added inadvertently due to the input having whitespace in between tags. 
The reason for this extra check and removal of tags / entries is because it was noticed during testing that our solution to disable whitespace trimming also disables trimming in the space between closing and opening tags. Some inputs are formated with whitespace to make them readable. The parser's behaviour in this situation is to put the whitespace in an array with the tag "content" (or whatver getcDataTagName() returns). As it would be a rather complicated change to the parser to change this, we opted to rather remove anything that was inadvertently added at this point in the processing. We have also added that using inputs that share the same name as cDataTagName is unsupported when whitespace trimming is disabled to avoid issues. There is an example of the incorrect output in the linked issue as well as a test case which asserts the behaviour of using a tag with the cDataTag name to show why it is unsupported